moveit_core/robot_model/src/planar_joint_model.cpp

```diff
@@ -174,7 +174,15 @@ void computeTurnDriveTurnGeometry(const double* from, const double* to, const do
     initial_turn = angle_backward_diff;
   }
   drive_angle = from[2] + initial_turn;
-  final_turn = angles::shortest_angular_distance(drive_angle, to[2]);
+  final_turn = to[2] - drive_angle;
+
+  // Check if normalization is needed, because
+  // if fabs(final_turn) << M_PI, the floating point arithmetic performed in `normalize_angle`
+  // will cause final_turn to be rounded to 0
+  if (final_turn < -boost::math::constants::pi<double>() || final_turn >= boost::math::constants::pi<double>())
+  {
+    final_turn = angles::normalize_angle(final_turn);
+  }
```

Comment:
As you correctly handle `final_turn == 0` below, why do you need to replace `angles::shortest_angular_distance`?

Reply:
When using `angles::shortest_angular_distance(drive_angle, to[2])`, if `to[2] - drive_angle << M_PI`, the result is `final_turn = 0.0`. 

Whereas with this implementation, `final_turn` will actually be assigned the difference between `to[2]` and `drive_angle` regardless of the magnitude of the difference. 

The reason this occurs is because the [`normalize()` function](https://github.com/ros/angles/blob/54ad72b0ce473a2e1d98da447898bf2fcd36519f/angles/include/angles/angles.h#L84C1-L89) does an addition by `M_PI` and then subtraction by `M_PI`, thus `normalize(x) != x` if `x` is very small due to floating point arithmetic, as shown by [this script](https://onlinegdb.com/XfEgGWwEYJ).

Note that we are performing the exact same operations as [`angles::shortest_angular_distance`](https://github.com/ros/angles/blob/54ad72b0ce473a2e1d98da447898bf2fcd36519f/angles/include/angles/angles.h#L104-L107), except with an extra check to see if normalization is actually necessary, in order to avoid this loss of precision.

I was not sure whether this loss of precision was acceptable, so I wrote it this way to avoid truncation as much as possible. I can revert it to `shortest_angular_distance` if this error is acceptable.